client/src/classes/item.ts

```diff
@@ -40,11 +44,31 @@ export class Item {
   public setAmount(_amount: number) : void { 
```

Comment:
Generally seems as though there's very strong coupling between Item and ItemTemplateStore in a way that expects Item to be responsible for the state of the template store.